doc/content/devices/adding-devices/_index.md

```diff
@@ -23,103 +23,103 @@ To create a device, first open the application you wish to add the device in. Go
+Enter a **JoinEUI/AppEUI** if provided by your manufacturer and click **Confirm**. If it is not provided by the manufacturer and your device is programmable, you can generate a random one in accordance with the test ranges defined by the [IEEE 802 standards](https://ieee802.org/) or use all zeros, just make sure to program the same value into your device. 
 
-Enter your **DevEUI**. This should be provided by your manufacturer for commercial devices. If your device is programmable, you may generate an EUI using the **Generate** button, and program it in your device.
+Now enter your **DevEUI**. This should be provided by your manufacturer for commercial devices. If your device is programmable, you may generate an EUI using the **Generate** button, and program it in your device.
 
-For LoRaWAN version 1.0.x devices, you will see an **AppKey** field. If your manufacturer provides an **AppKey**, enter it. Otherwise, use the **Generate** button to create one, and program it in to your device.
+{{< note >}} Note that the **JoinEUI** and **DevEUI** fields will be pre-filled if you scanned your device's QR code as explained in [Onboarding Devices using QR Codes]({{< ref "/devices/adding-devices#onboarding-devices-using-qr-codes" >}}) section. {{</ note >}}
 
-For LoRaWAN version 1.1.x devices, you will also see a **NwkKey** field. If it is provided by your manufacturer, enter it. Otherwise, use the **Generate** button to create one, and program it in to your device.
+For LoRaWAN version 1.0.x devices, you will see an **AppKey** field, and for LoRaWAN version 1.1.x devices you will also see a **NwkKey** field. If these keys are provided by your manufacturer, enter them. Otherwise, use the **Generate** button to create them, and program them into your device.
 
 {{< figure src="device-repo-settings.png" alt="Device information" >}}
 
 Finally, give your device a unique **End device ID**, and click the **Register end device** button to create the end device. See [ID and EUI constraints]({{< ref "reference/id-eui-constraints" >}}) for guidelines about choosing a unique ID.
 
-{{< figure src="device-repo-settings.png" alt="Device information" >}}
-
 The device is now activated, and will appear as connected in {{% tts %}} once it sends an uplink.
 
 ### Manually Registering an End Device
 
-If your device is not available in the device repository, you may manually register it.
+If your device is not available in the device repository, you may manually register it. To do this, choose the **Enter end device specifics manually** input method.
 
-Make sure the **Manually** tab is selected.
-
-{{< figure src="manual-tab.png" alt="Manual tab" >}}
+Choose a **Frequency plan** appropriate for your region. Your device and gateway must use the same frequency plan to communicate.
```

Comment:
Can we link any assisting info here as to what preferred FPs per regions are?

Reply:
https://www.thethingsnetwork.org/docs/lorawan/frequency-plans/
https://www.thethingsnetwork.org/docs/lorawan/frequencies-by-country/